hybrasyl/Messaging/Admin.cs

```diff
@@ -0,0 +1,410 @@
+                    }
+
+                    // Report to the end user
+                    return Success($"{scriptStatus}\n\n{errorSummary}", MessageTypes.SLATE_WITH_SCROLLBAR);
+                }
+            }
+            return Fail("Script {args[1].Trim()}: not found.");
+        }
+    }
+
+    class RollcharCommand : ChatCommand
+    {
+        public new static string Command = "rollchar";
+        public new static string ArgumentText = "<string class> <byte level>";
+        public new static string HelpText = "Simulate a new character with the specified class and level, and show resulting HP/MP.";
+        public new static bool Privileged = true;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+            if (!Enum.IsDefined(typeof(Enums.Class), args[1]))
+                return Fail($"Invalid class {args[0]}");
+
+            if (!byte.TryParse(args[2], out byte level) || level < 1 || level > Constants.MAX_LEVEL)
+                return Fail("Invalid level: must be between 1 and {Constants.MAX_LEVEL}");
+
+            // Create a fake User, and level them up to the desired level
+
+            var testUser = new User();
+            testUser.Map = new Map();
+            testUser.Stats.BaseHp = 50;
+            testUser.Stats.BaseMp = 50;
+            testUser.Class = (Enums.Class)Enum.Parse(typeof(Enums.Class), args[1]);
+            testUser.Stats.Level = 1;
+
+            while (testUser.Stats.Level < level)
+            {
+                testUser.GiveExperience(testUser.ExpToLevel);
+            }
+
+            return Success($"{testUser.Class}, Level {testUser.Stats.Level}, Hp: {testUser.Stats.BaseHp}, Mp: {testUser.Stats.BaseMp}");
+        }
+    }
+
+    class TeleportCommand : ChatCommand
+    {
+        public new static string Command = "teleport";
+        public new static string ArgumentText = "<string playername> | <uint mapnumber> <byte x> <byte y>";
+        public new static string HelpText = "Teleport to the specified player, or the given map number and coordinates.";
+        public new static bool Privileged = false;
+
+        public new static ChatCommandResult Run(User user, params string[] args)
+        {
+
+            if (args.Length == 1)
+            {
+                if (!Game.World.WorldData.ContainsKey<User>(args[1]))
+                {
+                    return Fail("That player cannot be found");
+                }
+                else
+                {
+                    var target = Game.World.WorldData.Get<User>(args[1]);
+                    user.Teleport(target.MapId, target.X, target.Y);
+                    return Success($"Teleported to {target.Name} - {target.Map.Name} ({target.X},{target.Y})");
+                }
+            }
+            else
+            {
+                if (ushort.TryParse(args[0], out ushort mapnum) && !byte.TryParse(args[1], out byte x) && !byte.TryParse(args[2], out byte y))
```

Comment:
Should remove the ! infront of byte. They are turning the successful TryParse into "false".

Reply:
fixed, tyop